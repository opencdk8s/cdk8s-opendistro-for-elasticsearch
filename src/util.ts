
import * as k8s from './imports/k8s';

export interface ResourceRequirements {
  /**
     * Maximum resources for the web app.
     * @default - CPU = 400m, Mem = 512Mi
     */
  readonly limits?: ResourceQuantity;

  /**
     * Required resources for the web app.
     * @default - CPU = 200m, Mem = 256Mi
     */
  readonly requests?: ResourceQuantity;
}

export interface ResourceQuantity {
  /**
   * @default - no limit
   */
  readonly cpu?: string;

  /**
   * @default - no limit
   */
  readonly memory?: string;
}

export function convertQuantity(
  user: ResourceQuantity | undefined,
  defaults: { cpu: string; memory: string },
): { [key: string]: k8s.Quantity } {
  // defaults
  if (!user) {
    return {
      cpu: k8s.Quantity.fromString(defaults.cpu),
      memory: k8s.Quantity.fromString(defaults.memory),
    };
  }

  const result: { [key: string]: k8s.Quantity } = {};

  if (user.cpu) {
    result.cpu = k8s.Quantity.fromString(user.cpu);
  }

  if (user.memory) {
    result.memory = k8s.Quantity.fromString(user.memory);
  }

  return result;
}