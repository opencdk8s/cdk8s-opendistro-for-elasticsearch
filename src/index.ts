import { Construct } from 'constructs';
import { MyClient } from './constructs/client';
import { MyData } from './constructs/data';
import { MyKibana } from './constructs/kibana';
import { MyMaster } from './constructs/master';
import { MyMisc } from './constructs/misc';
import { ResourceRequirements } from './util';
export * from './util';

export interface EsOpts {
  // Common
  readonly name?: string;
  readonly namespace?: string;
  readonly image?: string;
  // Master
  readonly masterVolumeSize?: string;
  readonly masterReplicas?: number;
  readonly masterResources?: ResourceRequirements;
  readonly masterNodeSelectorParams?: { [name: string]: string };
  // Client
  readonly clientVolumeSize?: string;
  readonly clientReplicas?: number;
  readonly clientResources?: ResourceRequirements;
  readonly clientNodeSelectorParams?: { [name: string]: string };
  // Data
  readonly dataVolumeSize?: string;
  readonly dataReplicas?: number;
  readonly dataResources?: ResourceRequirements;
  readonly dataNodeSelectorParams?: { [name: string]: string };
  // Kibana
  readonly kibanaImage?: string;
  readonly kibanaResources?: ResourceRequirements;
  readonly kibanaNodeSelectorParams?: { [name: string]: string };
  readonly kibanaReplicas?: number;
}

export class MyElasticSearch extends Construct {

  /**
   * Namespace
   *
   * @default - elasticsearch
   */
  public readonly kibanaImage?: string;
  public readonly name?: string;
  public readonly namespace?: string;
  public readonly image?: string;

  constructor(scope: Construct, name: string, opts: EsOpts) {
    super(scope, name);

    this.name = opts.name ?? 'elasticsearch';
    this.namespace = opts.namespace ?? 'elasticsearch';
    this.image = opts.image ?? 'docker.io/amazon/opendistro-for-elasticsearch:1.13.2';
    this.kibanaImage = opts.kibanaImage ?? 'docker.io/amazon/opendistro-for-elasticsearch-kibana:1.13.2';

    new MyMaster(this, `${this.name}-master`, {
      masterReplicas: opts.masterReplicas,
      masterVolumeSize: opts.masterVolumeSize,
      masterNodeSelectorParams: opts.masterNodeSelectorParams,
      masterResources: opts.masterResources,
      namespace: this.namespace,
      image: this.image,
      name: this.name,
    });

    new MyClient(this, `${this.name}-client`, {
      clientReplicas: opts.clientReplicas,
      clientVolumeSize: opts.clientVolumeSize,
      clientNodeSelectorParams: opts.clientNodeSelectorParams,
      clientResources: opts.clientResources,
      namespace: this.namespace,
      image: this.image,
      name: this.name,
    });

    new MyData(this, `${this.name}-data`, {
      dataReplicas: opts.dataReplicas,
      dataVolumeSize: opts.dataVolumeSize,
      dataNodeSelectorParams: opts.dataNodeSelectorParams,
      dataResources: opts.dataResources,
      namespace: this.namespace,
      image: this.image,
      name: this.name,
    });

    new MyMisc(this, `${this.name}`, {
      namespace: this.namespace,
    });

    new MyKibana(this, `${this.name}-kibana`, {
      kibanaReplicas: opts.kibanaReplicas,
      kibanaImage: opts.kibanaImage,
      kibanaNodeSelectorParams: opts.kibanaNodeSelectorParams,
      kibanaResources: opts.kibanaResources,
      namespace: this.namespace,
      name: this.name,
    });

  }
}