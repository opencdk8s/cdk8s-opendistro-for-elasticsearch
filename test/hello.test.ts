import { Chart, Testing } from 'cdk8s';
import * as es from '../src/index';

test('elasticsearch', () => {
  const app = Testing.app();
  const chart = new Chart(app, 'test');
  new es.MyElasticSearch(chart, 'es', {
    name: 'elasticsearch',
  });
  expect(Testing.synth(chart)).toMatchSnapshot();
});
