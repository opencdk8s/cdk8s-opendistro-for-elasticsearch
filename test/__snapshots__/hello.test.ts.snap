// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elasticsearch 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
        "role": "master",
      },
      "name": "elasticsearch-discovery",
      "namespace": "elasticsearch",
    },
    "spec": Object {
      "clusterIP": "None",
      "ports": Array [
        Object {
          "port": 9300,
          "protocol": "TCP",
        },
      ],
      "selector": Object {
        "role": "master",
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "StatefulSet",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
        "role": "master",
      },
      "name": "elasticsearch-master",
      "namespace": "elasticsearch",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "elasticsearch",
          "role": "master",
        },
      },
      "serviceName": "elasticsearch-discovery",
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "elasticsearch",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "cluster.name",
                  "value": "elasticsearch",
                },
                Object {
                  "name": "cluster.initial_master_nodes",
                  "value": "elasticsearch-master-0,",
                },
                Object {
                  "name": "node.master",
                  "value": "true",
                },
                Object {
                  "name": "node.ingest",
                  "value": "false",
                },
                Object {
                  "name": "node.data",
                  "value": "false",
                },
                Object {
                  "name": "network.host",
                  "value": "0.0.0.0",
                },
                Object {
                  "name": "node.name",
                  "valueFrom": Object {
                    "fieldRef": Object {
                      "fieldPath": "metadata.name",
                    },
                  },
                },
                Object {
                  "name": "discovery.seed_hosts",
                  "value": "elasticsearch-discovery",
                },
                Object {
                  "name": "KUBERNETES_NAMESPACE",
                  "valueFrom": Object {
                    "fieldRef": Object {
                      "fieldPath": "metadata.namespace",
                    },
                  },
                },
                Object {
                  "name": "PROCESSORS",
                  "valueFrom": Object {
                    "resourceFieldRef": Object {
                      "resource": "limits.cpu",
                    },
                  },
                },
                Object {
                  "name": "ES_JAVA_OPTS",
                  "value": "-Xms512m -Xmx512m",
                },
              ],
              "image": "docker.io/amazon/opendistro-for-elasticsearch:1.13.2",
              "imagePullPolicy": "Always",
              "livenessProbe": Object {
                "initialDelaySeconds": 60,
                "periodSeconds": 10,
                "tcpSocket": Object {
                  "port": "transport",
                },
              },
              "name": "elasticsearch",
              "ports": Array [
                Object {
                  "containerPort": 9300,
                  "name": "transport",
                },
                Object {
                  "containerPort": 9200,
                  "name": "http",
                },
                Object {
                  "containerPort": 9600,
                  "name": "metrics",
                },
                Object {
                  "containerPort": 9650,
                  "name": "rca",
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "400m",
                  "memory": "512Mi",
                },
                "requests": Object {
                  "cpu": "200m",
                  "memory": "256Mi",
                },
              },
              "securityContext": Object {
                "capabilities": Object {
                  "add": Array [
                    "SYS_CHROOT",
                  ],
                },
              },
              "volumeMounts": Array [
                Object {
                  "mountPath": "/usr/share/elasticsearch/data",
                  "name": "data",
                },
                Object {
                  "mountPath": "/usr/share/elasticsearch/config/logging.yml",
                  "name": "config",
                  "subPath": "logging.yml",
                },
              ],
            },
          ],
          "initContainers": Array [
            Object {
              "command": Array [
                "sysctl",
                "-w",
                "vm.max_map_count=262144",
              ],
              "image": "docker.io/busybox:1.27.2",
              "name": "init-sysctl",
              "securityContext": Object {
                "privileged": true,
              },
            },
            Object {
              "command": Array [
                "sh",
                "-c",
                "chown -R 1000:1000 /usr/share/elasticsearch/data",
              ],
              "image": "docker.io/busybox:1.27.2",
              "name": "fixmount",
              "volumeMounts": Array [
                Object {
                  "mountPath": "/usr/share/elasticsearch/data",
                  "name": "data",
                },
              ],
            },
          ],
          "serviceAccountName": "elasticsearch-es",
          "volumes": Array [
            Object {
              "name": "config",
              "secret": Object {
                "secretName": "elasticsearch-es-config",
              },
            },
          ],
        },
      },
      "updateStrategy": Object {
        "type": "RollingUpdate",
      },
      "volumeClaimTemplates": Array [
        Object {
          "metadata": Object {
            "name": "data",
          },
          "spec": Object {
            "accessModes": Array [
              "ReadWriteOnce",
            ],
            "resources": Object {
              "requests": Object {
                "storage": "8Gi",
              },
            },
          },
        },
      ],
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
        "role": "client",
      },
      "name": "elasticsearch-client-service",
      "namespace": "elasticsearch",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "name": "http",
          "port": 9200,
        },
        Object {
          "name": "transport",
          "port": 9300,
        },
        Object {
          "name": "metrics",
          "port": 9600,
        },
        Object {
          "name": "rca",
          "port": 9650,
        },
      ],
      "selector": Object {
        "role": "client",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
        "role": "client",
      },
      "name": "elasticsearch-client",
      "namespace": "elasticsearch",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "elasticsearch",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "elasticsearch",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "cluster.name",
                  "value": "elasticsearch",
                },
                Object {
                  "name": "node.master",
                  "value": "false",
                },
                Object {
                  "name": "node.ingest",
                  "value": "true",
                },
                Object {
                  "name": "node.data",
                  "value": "false",
                },
                Object {
                  "name": "network.host",
                  "value": "0.0.0.0",
                },
                Object {
                  "name": "node.name",
                  "valueFrom": Object {
                    "fieldRef": Object {
                      "fieldPath": "metadata.name",
                    },
                  },
                },
                Object {
                  "name": "discovery.seed_hosts",
                  "value": "elasticsearch-discovery",
                },
                Object {
                  "name": "KUBERNETES_NAMESPACE",
                  "valueFrom": Object {
                    "fieldRef": Object {
                      "fieldPath": "metadata.namespace",
                    },
                  },
                },
                Object {
                  "name": "PROCESSORS",
                  "valueFrom": Object {
                    "resourceFieldRef": Object {
                      "resource": "limits.cpu",
                    },
                  },
                },
                Object {
                  "name": "ES_JAVA_OPTS",
                  "value": "-Xms512m -Xmx512m",
                },
              ],
              "image": "docker.io/amazon/opendistro-for-elasticsearch:1.13.2",
              "imagePullPolicy": "Always",
              "livenessProbe": Object {
                "initialDelaySeconds": 60,
                "periodSeconds": 10,
                "tcpSocket": Object {
                  "port": "transport",
                },
              },
              "name": "elasticsearch",
              "ports": Array [
                Object {
                  "containerPort": 9200,
                  "name": "http",
                },
                Object {
                  "containerPort": 9300,
                  "name": "transport",
                },
                Object {
                  "containerPort": 9600,
                  "name": "metrics",
                },
                Object {
                  "containerPort": 9650,
                  "name": "rca",
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "400m",
                  "memory": "512Mi",
                },
                "requests": Object {
                  "cpu": "200m",
                  "memory": "256Mi",
                },
              },
              "securityContext": Object {
                "capabilities": Object {
                  "add": Array [
                    "SYS_CHROOT",
                  ],
                },
              },
              "volumeMounts": Array [
                Object {
                  "mountPath": "/usr/share/elasticsearch/config/logging.yml",
                  "name": "config",
                  "subPath": "logging.yml",
                },
              ],
            },
          ],
          "initContainers": Array [
            Object {
              "command": Array [
                "sysctl",
                "-w",
                "vm.max_map_count=262144",
              ],
              "image": "docker.io/busybox:1.27.2",
              "name": "init-sysctl",
              "securityContext": Object {
                "privileged": true,
              },
            },
          ],
          "serviceAccountName": "elasticsearch-es",
          "volumes": Array [
            Object {
              "name": "config",
              "secret": Object {
                "secretName": "elasticsearch-es-config",
              },
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
        "role": "data",
      },
      "name": "elasticsearch-data-svc",
      "namespace": "elasticsearch",
    },
    "spec": Object {
      "clusterIP": "None",
      "ports": Array [
        Object {
          "name": "transport",
          "port": 9300,
        },
        Object {
          "name": "http",
          "port": 9200,
        },
        Object {
          "name": "metrics",
          "port": 9600,
        },
        Object {
          "name": "rca",
          "port": 9650,
        },
      ],
      "selector": Object {
        "role": "data",
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "StatefulSet",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
        "role": "data",
      },
      "name": "elasticsearch-data",
      "namespace": "elasticsearch",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "elasticsearch",
          "role": "data",
        },
      },
      "serviceName": "elasticsearch-data-svc",
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "elasticsearch",
            "role": "data",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "cluster.name",
                  "value": "elasticsearch",
                },
                Object {
                  "name": "node.master",
                  "value": "false",
                },
                Object {
                  "name": "node.ingest",
                  "value": "false",
                },
                Object {
                  "name": "network.host",
                  "value": "0.0.0.0",
                },
                Object {
                  "name": "node.name",
                  "valueFrom": Object {
                    "fieldRef": Object {
                      "fieldPath": "metadata.name",
                    },
                  },
                },
                Object {
                  "name": "discovery.seed_hosts",
                  "value": "elasticsearch-discovery",
                },
                Object {
                  "name": "KUBERNETES_NAMESPACE",
                  "valueFrom": Object {
                    "fieldRef": Object {
                      "fieldPath": "metadata.namespace",
                    },
                  },
                },
                Object {
                  "name": "node.data",
                  "value": "true",
                },
                Object {
                  "name": "PROCESSORS",
                  "valueFrom": Object {
                    "resourceFieldRef": Object {
                      "resource": "limits.cpu",
                    },
                  },
                },
                Object {
                  "name": "ES_JAVA_OPTS",
                  "value": "-Xms512m -Xmx512m",
                },
              ],
              "image": "docker.io/amazon/opendistro-for-elasticsearch:1.13.2",
              "imagePullPolicy": "Always",
              "livenessProbe": Object {
                "initialDelaySeconds": 60,
                "periodSeconds": 10,
                "tcpSocket": Object {
                  "port": "transport",
                },
              },
              "name": "elasticsearch",
              "ports": Array [
                Object {
                  "containerPort": 9300,
                  "name": "transport",
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "400m",
                  "memory": "512Mi",
                },
                "requests": Object {
                  "cpu": "200m",
                  "memory": "256Mi",
                },
              },
              "securityContext": Object {
                "capabilities": Object {
                  "add": Array [
                    "SYS_CHROOT",
                  ],
                },
              },
              "volumeMounts": Array [
                Object {
                  "mountPath": "/usr/share/elasticsearch/data",
                  "name": "data",
                },
                Object {
                  "mountPath": "/usr/share/elasticsearch/config/logging.yml",
                  "name": "config",
                  "subPath": "logging.yml",
                },
              ],
            },
          ],
          "initContainers": Array [
            Object {
              "command": Array [
                "sysctl",
                "-w",
                "vm.max_map_count=262144",
              ],
              "image": "docker.io/busybox:1.27.2",
              "name": "init-sysctl",
              "securityContext": Object {
                "privileged": true,
              },
            },
            Object {
              "command": Array [
                "sh",
                "-c",
                "chown -R 1000:1000 /usr/share/elasticsearch/data",
              ],
              "image": "docker.io/busybox:1.27.2",
              "name": "fixmount",
              "volumeMounts": Array [
                Object {
                  "mountPath": "/usr/share/elasticsearch/data",
                  "name": "data",
                },
              ],
            },
          ],
          "serviceAccountName": "elasticsearch-es",
          "volumes": Array [
            Object {
              "name": "config",
              "secret": Object {
                "secretName": "elasticsearch-es-config",
              },
            },
          ],
        },
      },
      "updateStrategy": Object {
        "type": "RollingUpdate",
      },
      "volumeClaimTemplates": Array [
        Object {
          "metadata": Object {
            "name": "data",
          },
          "spec": Object {
            "accessModes": Array [
              "ReadWriteOnce",
            ],
            "resources": Object {
              "requests": Object {
                "storage": "8Gi",
              },
            },
          },
        },
      ],
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "labels": Object {
        "name": "elasticsearch",
      },
      "name": "elasticsearch",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
      },
      "name": "elasticsearch-es",
      "namespace": "elasticsearch",
    },
  },
  Object {
    "apiVersion": "v1",
    "data": Object {
      "logging.yml": "YXBwZW5kZXI6CiAgY29uc29sZToKICAgIGxheW91dDoKICAgICAgY29udmVyc2lvblBhdHRlcm46ICdbJWR7SVNPODYwMX1dWyUtNXBdWyUtMjVjXSAlbSVuJwogICAgICB0eXBlOiBjb25zb2xlUGF0dGVybgogICAgdHlwZTogY29uc29sZQplcy5sb2dnZXIubGV2ZWw6IElORk8KbG9nZ2VyOgogIGFjdGlvbjogREVCVUcKICBjb20uYW1hem9uYXdzOiBXQVJOCnJvb3RMb2dnZXI6ICR7ZXMubG9nZ2VyLmxldmVsfSwgY29uc29sZQ==",
    },
    "kind": "Secret",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
      },
      "name": "elasticsearch-es-config",
      "namespace": "elasticsearch",
    },
    "type": "Opaque",
  },
  Object {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
      },
      "name": "elasticsearch-kibana",
      "namespace": "elasticsearch",
    },
  },
  Object {
    "apiVersion": "policy/v1beta1",
    "kind": "PodSecurityPolicy",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
      },
      "name": "elasticsearch-psp",
      "namespace": "elasticsearch",
    },
    "spec": Object {
      "allowedCapabilities": Array [
        "SYS_CHROOT",
      ],
      "fsGroup": Object {
        "ranges": Array [
          Object {
            "max": 65535,
            "min": 1,
          },
        ],
        "rule": "MustRunAs",
      },
      "hostIPC": false,
      "hostNetwork": false,
      "hostPID": false,
      "privileged": true,
      "readOnlyRootFilesystem": false,
      "runAsUser": Object {
        "rule": "RunAsAny",
      },
      "seLinux": Object {
        "rule": "RunAsAny",
      },
      "supplementalGroups": Object {
        "ranges": Array [
          Object {
            "max": 65535,
            "min": 1,
          },
        ],
        "rule": "MustRunAs",
      },
      "volumes": Array [
        "configMap",
        "emptyDir",
        "projected",
        "secret",
        "downwardAPI",
        "persistentVolumeClaim",
      ],
    },
  },
  Object {
    "apiVersion": "rbac.authorization.k8s.io/v1beta1",
    "kind": "Role",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
      },
      "name": "elasticsearch-es",
      "namespace": "elasticsearch",
    },
    "rules": Array [
      Object {
        "apiGroups": Array [
          "extensions",
        ],
        "resourceNames": Array [
          "elasticsearch-ps",
        ],
        "resources": Array [
          "podsecuritypolicies",
        ],
        "verbs": Array [
          "use",
        ],
      },
    ],
  },
  Object {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "RoleBinding",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
      },
      "name": "elasticsearch-elastic-rolebinding",
    },
    "roleRef": Object {
      "apiGroup": "rbac.authorization.k8s.io",
      "kind": "Role",
      "name": "elasticsearch-es",
    },
    "subjects": Array [
      Object {
        "kind": "ServiceAccount",
        "name": "elasticsearch-es",
        "namespace": "elasticsearch",
      },
    ],
  },
  Object {
    "apiVersion": "rbac.authorization.k8s.io/v1beta1",
    "kind": "Role",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
      },
      "name": "elasticsearch-kibana",
      "namespace": "elasticsearch",
    },
    "rules": Array [
      Object {
        "apiGroups": Array [
          "extensions",
        ],
        "resourceNames": Array [
          "elasticsearch-ps",
        ],
        "resources": Array [
          "podsecuritypolicies",
        ],
        "verbs": Array [
          "use",
        ],
      },
    ],
  },
  Object {
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "RoleBinding",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
      },
      "name": "elasticsearch-kibana-rolebinding",
    },
    "roleRef": Object {
      "apiGroup": "rbac.authorization.k8s.io",
      "kind": "Role",
      "name": "elasticsearch-kibana",
    },
    "subjects": Array [
      Object {
        "kind": "ServiceAccount",
        "name": "elasticsearch-kibana",
        "namespace": "elasticsearch",
      },
    ],
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
        "role": "kibana",
      },
      "name": "elasticsearch-kibana-svc",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "name": "kibana-svc",
          "port": 443,
          "targetPort": 5601,
        },
      ],
      "selector": Object {
        "role": "kibana",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "elasticsearch",
        "role": "kibana",
      },
      "name": "elasticsearch",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "elasticsearch",
          "role": "kibana",
        },
      },
      "spec": Object {
        "containers": Array [
          Object {
            "env": Array [
              Object {
                "name": "CLUSTER_NAME",
                "value": "elasticsearch",
              },
              Object {
                "name": "ELASTICSEARCH_HOSTS",
                "value": "elasticsearch-client-service:9200",
              },
            ],
            "image": "docker.io/amazon/opendistro-for-elasticsearch-kibana:1.13.2",
            "imagePullPolicy": "Always",
            "name": "elasticsearch-kibana",
            "ports": Array [
              Object {
                "containerPort": 5601,
              },
            ],
            "resources": Object {
              "limits": Object {
                "cpu": "400m",
                "memory": "512Mi",
              },
              "requests": Object {
                "cpu": "200m",
                "memory": "256Mi",
              },
            },
          },
        ],
        "restartPolicy": "Always",
        "serviceAccountName": "elasticsearch-kibana",
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "elasticsearch",
            "role": "kibana",
          },
        },
      },
    },
  },
]
`;
